src/leiningen/git_info_edn.clj

```diff
@@ -1,13 +1,15 @@
 (ns leiningen.git-info-edn
-  (:require [clojure.java.shell :as sh])
-  (:import (java.util Date)))
+  (:require [clojure.string :as s]
+            [clojure.java.shell :as sh])
+  (:import (java.time Instant)))   ; Java 1.8+
 
 (defn git-info-edn
-  "adds git info to the resources directory"
+  "adds git info to the resources directory (which must already exist)"
   [& _]
-  (spit "resources/deploy-info.edn"
-    (pr-str
-      {:hash (:out (sh/sh "git" "rev-parse" "HEAD"))
-       :status (:out (sh/sh "git" "status"))
-       :date (Date.)}))
+  (let [git-hash    (s/trim (:out (sh/sh "git" "rev-parse" "HEAD")))
+        git-status  (s/trim (:out (sh/sh "git" "status")))
+        now-rfc3339 (str (Instant/now))]
+    (spit "resources/deploy-info.edn"
+      ; Note: we have to manually emit the map, because of the #inst tag
+      (str "{:hash " (pr-str git-hash) ", :status " (pr-str git-status) ", :data #inst " (pr-str now-rfc3339) "}")))
```

Comment:
why is `pr-str` called on the time here, after explicitly using str above?

Reply:
couldn't the creation of the inst via string ops happen inside the value, while emitting the rest of the map via standard pr-str?

Reply:
> why is pr-str called on the time here, after explicitly using str above?

Because the behaviour of `str` and `pr-str` are different.
e.g.
```
user=> (def my-str "hello")
#'user/my-str
user=> (str my-str)
"hello"
user=> (pr-str my-str)
"\"hello\""
```

We need the latter semantics in this case.

> couldn't the creation of the inst via string ops happen inside the value, while emitting the rest of the map via standard pr-str?

Yes it could. I personally think that approach is less readable, but also think your call (as library author) should be final on such matters, so don't have any problem whatsoever if you were to change it.

Reply:
What I'm saying is instead of
```
(str "{:hash " (pr-str git-hash) ... ":data #inst" (pr-str now-rfc3339) "}")
```

You could have
```
(pr-str {:hash git-hash ... :date (str "#inst " (pr-str now-rfc3339))})
```
with the same string as a result, but less complexity in the form

Reply:
after more consideration, here's a draft of my idea... https://github.com/noisesmith/lein-git-info-edn/pull/5/files

Reply:
and I pushed a second iteration, using reify plus print-method instead of a format string hack